boot/nuttx/main.c

```diff
@@ -0,0 +1,93 @@
+/****************************************************************************
+ * boot/nuttx/main.c
+ *
+ * Copyright (c) 2021 Espressif Systems (Shanghai) Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <stdio.h>
+
+#include <sys/boardctl.h>
+
+#include <bootutil/bootutil.h>
+#include <bootutil/image.h>
+
+#include "flash_map_backend/flash_map_backend.h"
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * do_boot
+ ****************************************************************************/
+
+static void do_boot(struct boot_rsp *rsp)
+{
+  struct flash_area *flash_area;
+  struct boardioc_boot_info_s info;
+  int area_id;
+
+  area_id = flash_area_id_from_image_offset(rsp->br_image_off);
+
+  flash_area_open(area_id, &flash_area);
+
+  printf("Booting from %s...\n", flash_area->fa_mtd_path);
```

Comment:
I think we need an assert here to avoid "flash_area" returning a NULL pointer!

Reply:
Nice suggestion. I've added an assertion for ensuring `flash_area_open` won't fail here.
Thus the `flash_area` pointer is also assured to be valid.